Src/PerformanceCollector/Shared/Implementation/QuickPulse/IQuickPulseServiceClient.cs

```diff
@@ -16,15 +17,33 @@ internal interface IQuickPulseServiceClient
         /// </summary>
         /// <param name="instrumentationKey">InstrumentationKey for which to submit data samples.</param>
         /// <param name="timestamp">Timestamp to pass to the server.</param>
+        /// <param name="configurationETag">Current configuration ETag that the client has.</param>
+        /// <param name="authApiKey">Authentication API key.</param>
+        /// <param name="configurationInfo">When available, the deserialized response data received from the server.</param>
         /// <returns><b>true</b> if data is expected, otherwise <b>false</b>.</returns>
-        bool? Ping(string instrumentationKey, DateTimeOffset timestamp);
+        bool? Ping(
```

Comment:
my concern is that parameter list is long, but not complete. As you said you also pass role instance for instance. 

I'd say you need multiple instances of this one to support scenario when one process runs multiple "roles". It will be a service fabric case.